### PR TITLE
npctalk script command accept <color>

### PR DESCRIPTION
**Pull request by sader1992**
* **Addressed Issue(s)**: none

* **Server Mode**: both

* **Description of Pull Request**: 

npc talk now accept 0xRRGGBB color
Example:
```cs
prontera,153,180,0	script	testtt	444,{
	npctalk "test",strnpcinfo(0),bc_all,0xFF130B;
end;
}
```
https://color.adobe.com/create/color-wheel/

### CODE REVIEW

File: src/map/script.cpp [hunk 1]
```diff
@@ -15048,12 +15048,16 @@ BUILDIN_FUNC(npctalk)
...
+	int color = 0;
 
 	if (script_hasdata(st, 3) && strlen(script_getstr(st,3)) > 0)
 		nd = npc_name2id(script_getstr(st, 3));
 	else
 		nd = (struct npc_data *)map_id2bl(st->oid);
 
+	if (script_hasdata(st, 5))
+		color = script_getnum(st, 5); //Color
+
```
Comment (secretdataz) [inline src/map/script.cpp:15060]:
```suggestion
	if (script_hasdata(st, 5)) {
             swapColorBits = true;
             color = script_getnum(st, 5);
	}
	
```

File: src/map/script.cpp [hunk 2]
```diff
@@ -15069,12 +15073,18 @@ BUILDIN_FUNC(npctalk)
 		}
 		safesnprintf(message, sizeof(message), "%s", str);
 		if (target != SELF)
-			clif_messagecolor(&nd->bl, color_table[COLOR_WHITE], message, false, target);
+			if (script_hasdata(st, 5))
+				clif_messagecolor(&nd->bl, color, message, true, target);
+			else
+				clif_messagecolor(&nd->bl, color_table[COLOR_WHITE], message, false, target);
```
Comment (secretdataz) [inline src/map/script.cpp:15079]:
```suggestion
			clif_messagecolor(&nd->bl, color, message, swapColorBits, target);
```

File: src/map/script.cpp [hunk 3]
```diff
@@ -15069,12 +15073,18 @@ BUILDIN_FUNC(npctalk)
...
 		else {
 			TBL_PC *sd = map_id2sd(st->rid);
 			if (sd == NULL)
 				return SCRIPT_CMD_FAILURE;
-			clif_messagecolor_target(&nd->bl, color_table[COLOR_WHITE], message, false, target, sd);
+			if (script_hasdata(st, 5))
+				clif_messagecolor_target(&nd->bl, color, message, true, target, sd);
+			else
+				clif_messagecolor_target(&nd->bl, color_table[COLOR_WHITE], message, false, target, sd);
```
Comment (secretdataz) [inline src/map/script.cpp:15087]:
```suggestion
			clif_messagecolor_target(&nd->bl, color, message, swapColorBits, target, sd);
```

**Comment by secretdataz** [comment]
Additional note, as we've discussed on Discord, the boolean that I added isn't really needed since white is 0xFFFFFF so swapping bits result in the same color.

File: doc/script_commands.txt [hunk 4]
```diff
@@ -6588,13 +6588,14 @@ Returns true if the command was executed on the other NPC successfully, false if
...
-*npctalk "<message>"{,"<NPC name>","<flag>"};
+*npctalk "<message>"{,"<NPC name>",<flag>{,<color>}};
 
 This command will display a message as if the NPC object running it was a player
 talking - that is, above their head and in the chat window.
 The display name of the NPC won't get appended in front of the message.
 If the <NPC name> option is given and not empty, then that NPC will display the message,
-else the attached NPC will display the message.
+else the attached NPC will display the message,
+The color format is in RGB (0xRRGGBB). The color is by default White
```
Comment (aleos89) [inline doc/script_commands.txt:6598]:
```suggestion
the color format is in RGB (0xRRGGBB). The color White by default.
```

**Pull request merged**